examples/grpc_dynamic_routing/envoy_config.json

```diff
@@ -1370,6 +1370,202 @@
                               }
                             }
                           },
+                          {
+                            "decorator": {
+                              "operation": "ingress UnknownHttpMethodForPath_/echo"
+                            },
+                            "directResponse": {
+                              "body": {
+                                "inlineString": "The current request is matched to the defined url template \"/echo\" but its http method is not allowed"
+                              },
+                              "status": 405
+                            },
+                            "match": {
+                              "path": "/echo"
+                            }
+                          },
+                          {
+                            "decorator": {
+                              "operation": "ingress UnknownHttpMethodForPath_/echo"
+                            },
+                            "directResponse": {
+                              "body": {
+                                "inlineString": "The current request is matched to the defined url template \"/echo\" but its http method is not allowed"
+                              },
+                              "status": 405
+                            },
+                            "match": {
+                              "path": "/echo/"
+                            }
+                          },
+                          {
+                            "decorator": {
+                              "operation": "ingress UnknownHttpMethodForPath_/echoreport"
+                            },
+                            "directResponse": {
+                              "body": {
+                                "inlineString": "The current request is matched to the defined url template \"/echoreport\" but its http method is not allowed"
+                              },
+                              "status": 405
+                            },
+                            "match": {
+                              "path": "/echoreport"
+                            }
+                          },
+                          {
+                            "decorator": {
+                              "operation": "ingress UnknownHttpMethodForPath_/echoreport"
+                            },
+                            "directResponse": {
+                              "body": {
+                                "inlineString": "The current request is matched to the defined url template \"/echoreport\" but its http method is not allowed"
+                              },
+                              "status": 405
+                            },
+                            "match": {
+                              "path": "/echoreport/"
+                            }
```

Comment:
Please note, for the exactMatch, we will add additional one with trailing slash for backward compatibility. Here, we use the user defined one(no trailing slash) for the both the original one and generated one, as they only know what they defined.